### PR TITLE
tarray has predispose callback

### PR DESCRIPTION
**Pull request by anporumb**

### CODE REVIEW

File: tests/tarray_2_int/tarray_2_int.c [hunk 1]
```diff
@@ -0,0 +1,129 @@
...
+    ASSERT_IS_NOT_NULL(cats->arr[0].name);
+    free(cats->arr[0].name);
+}
+
+/*some thread that prints a cat name and then releases its hold on the TARRAY(CAT)*/
+static int cat_thread(void* p)
+{
+    TARRAY(CAT) allCats = p;
+    //LogInfo("hello from thread %" PRIu32 ". I have the cat \"%s\"", ThreadAPI_GetCurrentThreadId(), allCats->arr[0].name);
+    TARRAY_ASSIGN(CAT)(&allCats, NULL);
```
Comment (dcristoloveanu) [inline tests/tarray_2_int/tarray_2_int.c:77]: >allCats [](http://example.com/codeflow?start=24&length=7)

Consider all_cats as casing.

File: tests/tarray_2_int/tarray_2_int.c [hunk 2]
```diff
@@ -0,0 +1,129 @@
...
+{
+    ASSERT_IS_NOT_NULL(cats->arr[0].name);
+    free(cats->arr[0].name);
+}
+
+/*some thread that prints a cat name and then releases its hold on the TARRAY(CAT)*/
+static int cat_thread(void* p)
+{
+    TARRAY(CAT) allCats = p;
+    //LogInfo("hello from thread %" PRIu32 ". I have the cat \"%s\"", ThreadAPI_GetCurrentThreadId(), allCats->arr[0].name);
```
Comment (dcristoloveanu) [inline tests/tarray_2_int/tarray_2_int.c:76]: >//LogInfo("hello from thread %" PRIu32 ". I have th [](http://example.com/codeflow?start=4&length=51)

Not needed if it is commented

File: tests/tarray_ut/tarray_ut.c [hunk 3]
```diff
@@ -60,6 +50,11 @@ TARRAY_TYPE_DECLARE(THANDLE(A_TEST));
 TARRAY_TYPE_DEFINE(THANDLE(A_TEST));
 
 
+#define ENABLE_MOCKS
+#include "umock_c/umock_c_prod.h"
+MOCKABLE_FUNCTION(, void, my_cleanup, TARRAY_TYPEDEF_NAME(uint32_t)*, var2);
```
Comment (dcristoloveanu) [inline tests/tarray_ut/tarray_ut.c:55]: >uint32_t [](http://example.com/codeflow?start=58&length=8)

Needs an stdint include

File: tests/tarray_2_int/tarray_2_int.c [hunk 4]
```diff
@@ -0,0 +1,129 @@
...
+    TARRAY_ASSIGN(CAT)(&allCats, NULL);
+    return 0;
+}
+
+#define N_TIMES 1000
+
+/* the following test SHARES an array of cats between this thread (the main thread) and two others thread for the pupose of hearding the cats */
```
Comment (dcristoloveanu) [inline tests/tarray_2_int/tarray_2_int.c:83]: >hearding [](http://example.com/codeflow?start=124&length=8)

herding

**Review by dcristoloveanu — APPROVED** [review]
:shipit:

File: tests/tarray_2_int/tarray_2_int.c (hunk 1 above)
Comment (anporumb) [reply tests/tarray_2_int/tarray_2_int.c:77]: Renamed.

File: tests/tarray_2_int/tarray_2_int.c (hunk 2 above)
Comment (anporumb) [reply tests/tarray_2_int/tarray_2_int.c:76]: Removed

File: tests/tarray_2_int/tarray_2_int.c (hunk 4 above)
Comment (anporumb) [reply tests/tarray_2_int/tarray_2_int.c:83]: Cat are herded now./

File: tests/tarray_ut/tarray_ut.c (hunk 3 above)
Comment (anporumb) [reply tests/tarray_ut/tarray_ut.c:55]: added

**Pull request merged**